azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/AzureStorageCheckpointLeaseManager.java

```diff
@@ -489,126 +586,136 @@ private Boolean acquireLeaseSync(AzureBlobLease lease) throws Exception
-    	{
-    		if (wasLeaseLost(se, lease.getPartitionId()))
-    		{
-    			retval = false;
-    		}
-    		else
-    		{
-    			throw se;
-    		}
-    	}
-    	
-    	return retval;
+	    	TRACE_LOGGER.info(LoggingUtils.withHostAndPartition(this.host, lease, "Renewing lease"));
+	    	
+	    	CloudBlockBlob leaseBlob = ((AzureBlobLease)lease).getBlob();
+	    	boolean retval = true;
+	    	
+	    	try
+	    	{
+	    		leaseBlob.renewLease(AccessCondition.generateLeaseCondition(lease.getToken()), this.renewRequestOptions, null);
+	    	}
+	    	catch (StorageException se)
+	    	{
+	    		if (wasLeaseLost(se, lease.getPartitionId()))
+	    		{
+	    			retval = false;
+	    		}
+	    		else
+	    		{
+	        		throw LoggingUtils.wrapException(se, EventProcessorHostActionStrings.RENEWING_LEASE);
+	    		}
+	    	}
+	    	
+	    	return retval;
+    	}, this.host.getExecutorService());
     }
 
     @Override
-    public Future<Boolean> releaseLease(Lease lease)
-    {
-        return this.host.getExecutorService().submit(() -> releaseLeaseSync((AzureBlobLease)lease));
-    }
-    
-    private Boolean releaseLeaseSync(AzureBlobLease lease) throws Exception
+    public CompletableFuture<Boolean> releaseLease(Lease lease)
     {
-    	TRACE_LOGGER.info(LoggingUtils.withHostAndPartition(this.host.getHostName(), lease.getPartitionId(), "Releasing lease"));
-    	
-    	CloudBlockBlob leaseBlob = lease.getBlob();
-    	boolean retval = true;
-    	try
+    	return CompletableFuture.supplyAsync(() ->
     	{
-    		String leaseId = lease.getToken();
-    		AzureBlobLease releasedCopy = new AzureBlobLease(lease);
-    		releasedCopy.setToken("");
-    		releasedCopy.setOwner("");
-    		uploadLease(releasedCopy, leaseBlob, AccessCondition.generateLeaseCondition(leaseId), UploadActivity.Release, this.leaseOperationOptions);
-    		leaseBlob.releaseLease(AccessCondition.generateLeaseCondition(leaseId));
-    	}
-    	catch (StorageException se)
-    	{
-    		if (wasLeaseLost(se, lease.getPartitionId()))
-    		{
-    			retval = false;
-    		}
-    		else
-    		{
-    			throw se;
-    		}
-    	}
-    	
-    	return retval;
+	    	TRACE_LOGGER.info(LoggingUtils.withHostAndPartition(this.host, lease, "Releasing lease"));
+	    	
+	    	AzureBlobLease inLease = (AzureBlobLease)lease;
+	    	CloudBlockBlob leaseBlob = inLease.getBlob();
+	    	boolean retval = true;
+	    	try
+	    	{
+	    		String leaseId = lease.getToken();
+	    		AzureBlobLease releasedCopy = new AzureBlobLease(inLease);
+	    		releasedCopy.setToken("");
+	    		releasedCopy.setOwner("");
+	    		uploadLease(releasedCopy, leaseBlob, AccessCondition.generateLeaseCondition(leaseId), UploadActivity.Release, this.leaseOperationOptions);
+	    		leaseBlob.releaseLease(AccessCondition.generateLeaseCondition(leaseId));
+	    	}
+	    	catch (StorageException se)
+	    	{
+	    		if (wasLeaseLost(se, lease.getPartitionId()))
+	    		{
+	    			retval = false;
```

Comment:
>retval = false; [](start = 8, length = 15)

why ?
isn't intent of ReleaseLease fulfilled ? Shouldn't this be true ?

Reply:
Changed. With this change, releaseLease would only either return true or throw, so changed it to return Void and made appropriate upstream change.